hadoop/src/main/java/org/apache/iotdb/tsfile/hadoop/TSFRecordReader.java

```diff
@@ -155,58 +162,65 @@ public NullWritable getCurrentKey() throws IOException, InterruptedException {
   }
 
   @Override
-  public ArrayWritable getCurrentValue() throws IOException, InterruptedException {
-
-    Writable[] writables = getEmptyWritables();
-    Text deviceid = new Text(deviceId);
+  public MapWritable getCurrentValue() throws IOException, InterruptedException {
+    MapWritable mapWritable = new MapWritable();
+    Text deviceIdText = new Text(deviceIdList.get(currentIndex));
     LongWritable time = new LongWritable(timestamp);
     int index = 0;
-    if (isReadTime && isReadDeviceId) {
-      writables[0] = time;
-      writables[1] = deviceid;
-      index = 2;
-    } else if (isReadTime && !isReadDeviceId) {
-      writables[0] = time;
-      index = 1;
-    } else if (!isReadTime && isReadDeviceId) {
-      writables[0] = deviceid;
-      index = 1;
+
+    if (isReadTime && isReadDeviceId) { // Both time and deviceId need to be written into value
+      mapWritable.put(new Text("timestamp"), time);
+      mapWritable.put(new Text("device_id"), deviceIdText);
+    } else if (isReadTime) { // Only Time needs to be written into value
+      mapWritable.put(new Text("timestamp"), time);
+    } else if (isReadDeviceId) { // Only deviceId need to be written into value
+      mapWritable.put(new Text("device_id"), deviceIdText);
     }
```

Comment:
This might be simplified with
```
if (isReadTime) {
   mapWritable.put(new Text("timestamp"), time);
 }
 if (isReadDeviceId) { 
   mapWritable.put(new Text("device_id"), deviceIdText);
 }
```